creusot/src/translation/fmir.rs

```diff
@@ -213,9 +213,25 @@ impl<'tcx> Branches<'tcx> {
     }
 }
 
+#[derive(Debug, Clone, Copy)]
+pub enum InvariantKind {
+    /// Invariant that the user wrote
+    Source,
+    /// Invariant generated by desugaring `for` loops
+    For,
+    /// Invariant for mutated variables with type invariants
```

Comment:
Is this really what this kind is used for? Looking at its usage, I thought this was about mutable variables that are not mutated in the loop. Are type invariants involved?

Reply:
I misread the comment on `infer_proph_invariants`. It looks like it's about remembering the prophecies of mutable borrows, and doesn't depend on whether they are being mutated (e.g., test `while_let.rs`)